### PR TITLE
gnupg2: update to version 2.4.4

### PR DESCRIPTION
**Pull request by wteiken**
* removes patches for 2.4.3 as these are present in 2.4.4.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 14.3 23D56 arm64
Xcode 15.2 15C500b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: mail/gnupg2/Portfile [hunk 1]
```diff
@@ -4,14 +4,14 @@ PortSystem          1.0
 
 set my_name         gnupg
 name                ${my_name}2
-version             2.4.3
+version             2.4.4
 revision            1
```
Comment (reneeotten) [inline mail/gnupg2/Portfile:8]: revision should be reset to zero when updating the version

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: mail/gnupg2/Portfile (hunk 1 above)
Comment (wteiken) [reply mail/gnupg2/Portfile]: Done

File: mail/gnupg2/Portfile [hunk 2]
```diff
@@ -33,11 +33,6 @@ master_sites        ${my_name}:${my_name}
 use_bzip2           yes
 default_variants    +openldap
 
-# See: https://dev.gnupg.org/T6579
-patchfiles-append   patch-fix-build-without-openldap.diff
-# See: https://dev.gnupg.org/T6481
-patchfiles-append   patch-fix-emacs-easypg.diff
```
Comment (ryandesign) [inline mail/gnupg2/Portfile:39]: Please delete these files from the files directory.

File: mail/gnupg2/Portfile [hunk 3]
```diff
@@ -4,14 +4,14 @@ PortSystem          1.0
...
+version             2.4.4
+revision            0
 
 # Putting the SHA1 hash because that is published on the website
-checksums           rmd160  390bc6ae2011fa691f09cc65f0dee38c6ca73f2f \
-                    sha1    79a60c8e415e3daaa33d0546398174252a56f7ac \
-                    sha256  a271ae6d732f6f4d80c258ad9ee88dd9c94c8fdc33c3e45328c4d7c126bd219d \
-                    size    7351327
+checksums           rmd160  eab66d5a87e6e99cbf0586c289f13c85a4bf987f \
+                    sha1    228b3984325fdeebc5e3f2d165c6419a5ebc28de \
```
Comment (herbygillot) [inline mail/gnupg2/Portfile:12]: Also `sha1` is an obsolete checksum type. You can remove this line completely.

**Comment by pmetzger** [comment]
The commit message is garbled (esp. the name of the port.)

**Comment by pmetzger** [comment]
I believe @ryandesign and @reneeotten's issues have been resolved so I'm committing. Thanks, @wteiken!

**Pull request merged**